dbt_meshify/storage/file_content_editors.py

```diff
@@ -0,0 +1,453 @@
+
+        groups_yml["groups"] = list(groups.values())
+        return groups_yml
+
+    @staticmethod
+    def add_access_to_model_yml(
+        model_name: str, access_type: AccessType, models_yml: Dict[str, Any]
+    ):
+        """Add group and access configuration to a model's YAMl properties."""
+        # parse the yml file into a dictionary with model names as keys
+        models = resources_yml_to_dict(models_yml)
+        model_yml = models.get(model_name) or {"name": model_name, "columns": [], "config": {}}
+
+        model_yml.update({"access": access_type.value})
+        models[model_name] = process_model_yml(model_yml)
+
+        models_yml["models"] = list(models.values())
+        return models_yml
+
+    @staticmethod
+    def add_group_to_model_yml(model_name: str, group: Group, models_yml: Dict[str, Any]):
+        """Add group and access configuration to a model's YAMl properties."""
+        # parse the yml file into a dictionary with model names as keys
+        models = resources_yml_to_dict(models_yml)
+        model_yml = models.get(model_name) or {"name": model_name, "columns": [], "config": {}}
+
+        model_yml.update({"group": group.name})
+        models[model_name] = process_model_yml(model_yml)
+
+        models_yml["models"] = list(models.values())
+        return models_yml
+
+    @staticmethod
+    def add_group_and_access_to_model_yml(
+        model_name: str, group: Group, access_type: AccessType, models_yml: Dict[str, Any]
+    ):
+        """Add group and access configuration to a model's YAMl properties."""
+        # parse the yml file into a dictionary with model names as keys
+        models_yml = DbtMeshFileEditor.add_access_to_model_yml(model_name, access_type, models_yml)
+        models_yml = DbtMeshFileEditor.add_group_to_model_yml(model_name, group, models_yml)
+        return models_yml
+
+    def get_source_yml_entry(
+        self, resource_name: str, full_yml: Dict[str, Any], source_name: str
+    ) -> Tuple[Dict[str, Any], Optional[Dict[str, Any]]]:
+        """
+        Remove a single source entry from a source defintion block, return source definition with single source entry and the remainder of the original
+        """
+        sources = resources_yml_to_dict(full_yml, NodeType.Source)
+        source_definition = sources.get(source_name)
+        tables = source_definition.get("tables", None)
+        table = list(filter(lambda x: x["name"] == resource_name, tables))
+        remaining_tables = list(filter(lambda x: x["name"] != resource_name, tables))
+        resource_yml = source_definition.copy()
+        resource_yml["tables"] = table
+        source_definition["tables"] = remaining_tables
+        sources[source_name] = source_definition
+        if len(remaining_tables) == 0:
+            return resource_yml, None
+        else:
+            full_yml["sources"] = list(sources.values())
+            return resource_yml, full_yml
```

Comment:
The `else` is not required here.
```suggestion
        if len(remaining_tables) == 0:
            return resource_yml, None
 
        full_yml["sources"] = list(sources.values())
        return resource_yml, full_yml
```